devito/symbolics/inspection.py

```diff
@@ -7,18 +7,14 @@
 from devito.logger import warning
 from devito.tools import flatten
 
-
 def count(exprs, query):
     """
     Return a mapper ``{(k, v)}`` where ``k`` is a sub-expression in ``exprs``
     matching ``query`` and ``v`` is the number of its occurrences.
     """
-    mapper = OrderedDict()
+    mapper = {}
```

Comment:
Random guess.
Should this be a ``Counter`` too -- thus explaining why some tests are failing? The failures are due to capturing fewer common sub-expressions, which results in higher flop counts than what the tests are expecting. 

It's like with this solution, we get different ``hit``s here:
https://github.com/opesci/devito/blob/master/devito/dse/manipulation.py#L110
A ``hit`` is a candidate common sub-expression that gets replaced by a temporary